lib/ex_doc/retriever.ex

```diff
@@ -382,14 +382,24 @@ defmodule ExDoc.Retriever do
     {{kind, name, arity}, anno, _, doc, metadata} = callback
     actual_def = actual_def(name, arity, kind)
     doc_line = anno_line(anno)
-    annotations = annotations_from_metadata(metadata)
 
-    {:attribute, anno, :callback, {^actual_def, specs}} =
-      Enum.find(module_data.abst_code, &match?({:attribute, _, :callback, {^actual_def, _}}, &1))
+    specs =
+      case Map.fetch(module_data.callbacks, actual_def) do
+        {:ok, specs} ->
+          specs
+
+        :error ->
+          # gen_statem:state_name/3 is stored as StateName/3 in the docs chunk
+          name = name |> Atom.to_string() |> Macro.underscore() |> String.to_atom()
+          Map.fetch!(module_data.callbacks, {name, arity})
+      end
 
+    {:type, anno, _, _} = hd(specs)
```

Comment:
I have a silly question about the best way to use and references when using Kernel functions... so my question is it's okay to call only the function and avoid doing a full call like `Kernel.hd(specs)`?

Reply:
yes, it's totally ok to skip `Kernel.` since it's automatically imported everywhere.

Reply:
so the only reason to use a full declaration is when using pipes

Reply:
This is unrelated to pipes, you'd use full declaration when you're calling function from another module. You can skip the module name if the function is local to the current module or was imported using [`import/2`](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#import/2). `Kernel` and `Kernel.SpecialForms` are automatically imported to every module, that's why you can call e.g. `hd(list)` instead of `Kernel.hd(list)`.